plugins/woocommerce/tests/php/includes/shipping/flat-rate/class-wc-shipping-flat-rate-test.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+// phpcs:disable Squiz.Classes.ClassFileName.NoMatch, Squiz.Classes.ValidClassName.NotCamelCaps -- backcompat nomenclature.
+
+/**
+ * Test for WC_Shipping_Flat_Rate class.
+ */
+class WC_Shipping_Flat_Rate_Test extends WC_Unit_Test_Case {
+
+	/**
+	 * @var WC_Shipping_Flat_Rate Shipping method instance.
+	 */
+	private $sut;
+
+	/**
+	 * @var Closure Function to call protected method evaluate_cost.
+	 */
+	private $call_evaluate_cost;
+
+	/**
+	 * Set up test case.
+	 *
+	 * @return void
+	 */
+	public function setUp(): void {
+		parent::setUp();
+		$this->sut                = new WC_Shipping_Flat_Rate();
+		$this->call_evaluate_cost = function ( $sum, $args ) {
+			return $this->evaluate_cost( $sum, $args );
+		};
+		update_option( 'woocommerce_price_decimal_sep', ',' );
+		update_option( 'woocommerce_price_thousand_sep', '.' );
+	}
+
+	/**
+	 * Tear down test case.
+	 *
+	 * @return void
+	 */
+	public function tearDown(): void {
+		update_option( 'woocommerce_price_decimal_sep', '.' );
+		update_option( 'woocommerce_price_thousand_sep', ',' );
+		parent::tearDown();
+	}
+
+
+	/**
+	 * @testDox Shipping cost with decimal seperator works as expected.
+	 */
+	public function test_evaluate_cost_sep_dec() {
+		$val = $this->call_evaluate_cost->call(
+			$this->sut,
+			'12345,67',
+			array(
+				'qty'  => 1,
+				'cost' => 1,
+			)
+		);
+		$this->assertEquals( 12345.67, $val );
+	}
+
+	/**
+	 * @testDox Shipping cost with incorrect decimal seperator works as expected.
+	 */
+	public function test_evaluate_cost_dec_seperator_inverse() {
+		$val = $this->call_evaluate_cost->call(
+			$this->sut,
+			'12345.67',
+			array(
+				'qty'  => 1,
+				'cost' => 1,
+			)
+		);
+		$this->assertEquals( 12345.67, $val );
+	}
+
+	/**
+	 * @testDox Shipping cost with thousand and decimal seperator works as expected.
+	 */
+	public function test_evaluate_cost_sep_thou_dec() {
+		$this->markTestSkipped( 'This test currently fails because we dont support thousand seperator in shipping price.' );
```

Comment:
(There's a couple more tests using the same skipped message)

```suggestion
		$this->markTestSkipped( 'This test currently fails because we don\'t support thousand separator in shipping price.' );
```